src/eckit/utils/RLE.cc

```diff
@@ -104,44 +104,45 @@ long long RLEencode2timeout(T first, T last, U output, long long maxLoop,
             other += x;
         }
 
-        long long n = RLEencode2timeout(first, from, output, maxLoop, deadline);
+        long long n = RLEencode2timeout(first, from, output, maxLoop, deadline, depth > 0 ? depth - 1 : -1);
 
         if (k > 1) {
             *output++ = -k;
             n++;
             int m = RLEencode2timeout(from, from + x, make_dummy((typename std::iterator_traits<T>::value_type*)(0)),
-                                      maxLoop, deadline);
+                                      maxLoop, deadline, depth > 0 ? depth - 1 : -1);
```

Comment:
Can the logic be written as depth > 0 as normal, depth == 0 as corner case? This might be linked to maxLoop but, I don't see a reason to handle -1 AND 0 specially.